updates to GSoC Page after selection into the program (#3690)
kubeflow/website: content/en/events/gsoc-2024.md

---

Kubeflow Community is excited to announce that we have been [selected](https://summerofcode.withgoogle.com/programs/2024/organizations/kubeflow) as organization to participate in [**Google Summer of Code 2024**](https://buildyourfuture.withgoogle.com/programs/summer-of-code).

> Students, welcome to an incredible chance to dive into the vibrant Kubeflow community! Join us and become a valued contributor to our rapidly expanding network dedicated to AI/ML technologies on Kubernetes. Kubeflow mentors come with years of valuable experience and a passion for community growth and project enhancement, they're the guiding stars of our program. Immerse yourself in a dynamic environment where you'll not only gain insights from top industry experts but also sharpen your skills in open source collaboration, time management, and presentation prowess alongside making impactful technical contributions to projects of your interest. Our dedicated mentors will be by your side every step of the way, offering invaluable guidance and support throughout the project. Get ready to embark on an exciting journey of growth and learning!

Author: removed, and also accepted the following suggestions! Thank you.

> Below are the "ideas/features" from different projects in the Kubeflow that being submitted for seeking proposals from GSoC Students.

Reviewer: ```suggestion
Below are the "ideas/features" from different projects in Kubeflow that are seeking proposals from GSoC Students.
```

> ## How can I participate in Kubeflow GSoc?

Reviewer: Let's style as `GSoC` (different caps):

```suggestion
## How can I participate in Kubeflow GSoC?
```

> Please go to [Google Summer of Code 2024](https://buildyourfuture.withgoogle.com/programs/summer-of-code) and sign up as student and look up Kubeflow organization and select the projects you want to submit proposals. Note, you can submit more than one proposal. Contributor Applications open **March 18th, 2024** and end on **Apr 2nd 2024**. First read through contributor guidelines, how to participate GSoC google website. Only ask questions about projects with your mentors. General questions about GSoC please reach out to GSoC organizers.

Reviewer: Reworded:

```suggestion
Please go to [Google Summer of Code 2024](https://buildyourfuture.withgoogle.com/programs/summer-of-code) and sign up as a student. Next, look at the [projects below](#project-ideas-for-2024-gsoc) to decide which ones you are interested in. Note, that you must submit your proposals through the GSoC website and your proposal must be selected to participate.

Contributor applications open __March 18th, 2024__ and close on __April 2nd, 2024__. 

For more information, see the GSoC website and/or reach out to the GSoC organizers.
Please only contact mentors about projects, not the program itself.
```

> ## Slack

Reviewer: Let's make this H3, so it goes under `How can I participate in Kubeflow GSoc?`:

```suggestion
### Slack
```

If you need to reach out our mentors, join [Kubeflow Slack](https://www.kubeflow.org/docs/about/community/#kubeflow-slack) and join #gsoc-participants channel. Look up your mentor or find project specific channel on Kubeflow slack. Be kind to your mentor, search and see if the question has been already answered in the channel first. Note that for most projects there are more than one mentor you can tag and request help.

## Meetings

> To facilitate the questions from students, we ask you to find the next community meeting for the project you are interested in from below list and attend that meeting and ask questions.

Reviewer: Reworded:

```suggestion
You may wish to attend the next community meeting for the group that is leading your chosen project, please see the calendar below for more information.
```

<div id="calendar-container"></div>
<script type="text/javascript">
const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
const calender_src_list = [
  // Kubeflow Community
  "kubeflow.org_7l5vnbn8suj2se10sen81d9428%40group.calendar.google.com",
  // KServe Community
  "4fqdmu5fp4l0bgdlf4lm1atnsl2j4612%40import.calendar.google.com",
];
let calender_src = calender_src_list.map(src => `&src=${src}&color=%23A79B8E`).join('');
const html = `<iframe src="https://calendar.google.com/calendar/embed?ctz=${timezone}&height=600&wkst=1&bgcolor=%23ffffff&showPrint=0&showDate=1&mode=AGENDA&showTitle=0${calender_src}" style="border:solid 1px #777" width="800" height="600" frameborder="0" scrolling="no"></iframe>`;
document.getElementById('calendar-container').innerHTML = html;
</script>

> ## Didn't get selected for GSoC? Don't worry, you can still contribute!

Reviewer: This looks glitchy on the website, lets make it shorter:

```suggestion
## What if my proposal is not chosen?
```

> We understand that not everyone gets selected for Google Summer of Code (GSoC). However, we're still open to collaboration with enthusiastic students! While we can't offer a stipend like GSoC, we can provide valuable mentorship and experience.

Reviewer: ```suggestion
Please understand that not everyone can be selected for Google Summer of Code (GSoC), there are many possible candidates for each project.
However, we want to encourage you to participate in the Kubeflow project!
The best way to get started is to attend [working group meetings](/docs/about/community/#kubeflow-community-calendars) for components you want to contribute to.
```

> To ensure successful mentoring, we expect a strong commitment and enthusiasm from you for the chosen project. This way, our mentors can dedicate their time and expertise to guide you effectively.
> 
> We believe there are many ways to contribute outside of GSoC. If you're interested, please reach out to us!

Reviewer: This is not really related to the section, let's remove it:

```suggestion
```

> # Project Ideas for the 2024

Reviewer: This was incorrect English: 

```suggestion
# Project Ideas for 2024 GSoC
```

---

Kubeflow Community is excited to announce that we have been [selected](https://summerofcode.withgoogle.com/programs/2024/organizations/kubeflow) as organization to participate in [**Google Summer of Code 2024**](https://buildyourfuture.withgoogle.com/programs/summer-of-code).

> 

Reviewer: Let's remove the newlines, to make it a sentence. 

```suggestion
```

> Below are the "ideas/features" from different projects in Kubeflow that are seeking proposals from GSoC Students

Reviewer: Its probably not necessary to say this, and putting things in quotes looks a bit strange.

```suggestion
This page aims to help you participate in the Kubeflow organization for GSoC 2024.
```

## How can I participate in Kubeflow GSoC?

Please go to [Google Summer of Code 2024](https://buildyourfuture.withgoogle.com/programs/summer-of-code) and sign up as a student. Next, look at the [projects below](#project-ideas-for-2024-gsoc) to decide which ones you are interested in. Note, that you must submit your proposals through the GSoC website and your proposal must be selected to participate.

Contributor applications open __March 18th, 2024__ and close on __April 2nd, 2024__.

> 

Reviewer: Let's remove the newlines, to make it a sentence. 

```suggestion
```

For more information, see the GSoC website and/or reach out to the GSoC organizers.

> 

Reviewer: Let's remove the newlines, to make it a sentence. 

```suggestion
```

Please only contact mentors about projects, not the program itself.

### Slack

Please [__Join the Kubeflow Slack__](https://www.kubeflow.org/docs/about/community/#kubeflow-slack).

> Please __do not__ reach out privately to mentors, instead, start a thread in the `#gsoc-participants` channel so others can see the response. Be kind to our mentors, please search to see if your question has already been answered.

Reviewer 2: Would be nice to have Slack link here.
```suggestion
Please __do not__ reach out privately to mentors, instead, start a thread in the [`#gsoc-participants`](https://kubeflow.slack.com/archives/C06LW6Z3RA6) channel so others can see the response. Be kind to our mentors, please search to see if your question has already been answered.
```

Reviewer: agree

Author: updated now.

### Meetings

You may wish to attend the next community meeting for the group that is leading your chosen project, please see the calendar below for more information.

<div id="calendar-container"></div>
<script type="text/javascript">
const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
const calender_src_list = [
  // Kubeflow Community
  "kubeflow.org_7l5vnbn8suj2se10sen81d9428%40group.calendar.google.com",
  // KServe Community
  "4fqdmu5fp4l0bgdlf4lm1atnsl2j4612%40import.calendar.google.com",
];
let calender_src = calender_src_list.map(src => `&src=${src}&color=%23A79B8E`).join('');
const html = `<iframe src="https://calendar.google.com/calendar/embed?ctz=${timezone}&height=600&wkst=1&bgcolor=%23ffffff&showPrint=0&showDate=1&mode=AGENDA&showTitle=0${calender_src}" style="border:solid 1px #777" width="800" height="600" frameborder="0" scrolling="no"></iframe>`;
document.getElementById('calendar-container').innerHTML = html;
</script>

> ## What if my proposal is not chosen?

Reviewer 2: Should we section around contributing guides ?
For example:
Good starting point to learn how to contribute to Kubeflow is to follow [this guide](https://www.kubeflow.org/docs/about/contributing/).
Kubeflow components have individual contributing guide to understand the component better. For example:
- Katib [Contributing Guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md).
- Training Operator [Contributor Guide](https://github.com/kubeflow/training-operator/blob/master/docs/development/developer_guide.md).

Usually, issues with `good first issue` label is beginner friendly tasks that you can start to work on:

- Kubeflow Manifests [good first issues](https://github.com/kubeflow/manifests/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).
- Training Operator [good first issues](https://github.com/kubeflow/training-operator/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
- Katib [good first issues](https://github.com/kubeflow/katib/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).

Feel free to assign issues to yourself if you want to work on it, leave this comment in the issue `/assign @username`

Reviewer 2: cc @juliusvonkohout for review

Reviewer: This might be a little distracting on the page, perhaps we should just add a new section (with this info) to the top of the main contributor guide, and then link it here.

https://www.kubeflow.org/docs/about/contributing/

Please understand that not everyone can be selected for Google Summer of Code (GSoC), there are many possible candidates for each project.

> However, we want to encourage you to participate in the Kubeflow project!
> 
> The best way to get started is to attend [working group meetings](/docs/about/community/#kubeflow-community-calendars) for components you want to contribute to.

Reviewer: Removed the newline, and slightly reworded:

```suggestion
However, we still want to encourage you to participate in the Kubeflow project!
Get started by attending [working group meetings](/docs/about/community/#kubeflow-community-calendars) for components you want to help with, and reading our [contributing guide](/docs/about/contributing/).
```

**Mentors:** Andrey Velichkevich, Johnu George, Shivay Lamba, Yuan (Terry) Tang, Yuki Iwai

> **Component:** Katib/General

Reviewer 2: ```suggestion
**Component:** Katib/Training Operator
```
